### PR TITLE
fix(fsfreeze): improved error message when volume is not staged

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
### Describe the issue
It its seen that if a node has supposedly recovered from an abrupt shutdown and the application is not running, in that case the the filesystem mayastor volume is published but it's not mounted on the node where the application is. Mayastor control-plane volume being published gives the csi-controller a wrong notion that the volume is under use i.e mounted thus it tries to issue fsfreeze to ensure fs consistency, but fsfreeze wouldn't succeed as it would search for the mount path to issue FIFREEZE ioctl. In such case the flow returns a very vague error of `Terminated with error error=Cannot find volume: volume ID: e0316b8c-e4f4-454c-9803-c238dc68457c`.

### Changes
This PR enhances the error message stating what is the actual error i.e `Failed to freeze volume as the filesystem volume is not mounted`

### CODE REVIEW

File: control-plane/csi-driver/src/bin/controller/controller.rs [hunk 1]
```diff
@@ -861,10 +861,19 @@ impl rpc::csi::controller_server::Controller for CsiControllerSvc {
...
+                        issue_fs_freeze(app_node_endpoint.clone(), volume_uuid.to_string()).await
+                    {
+                        return match error.code() {
+                            Code::NotFound => Err(Status::not_found(format!(
+                                "Failed to freeze volume {}, filesystem volume is not mounted",
+                                volume_uuid
+                            ))),
+                            _ => Err(error),
+                        };
+                    }
```
Comment (tiagolobocastro) [inline control-plane/csi-driver/src/bin/controller/controller.rs]: I think only 1 match suffices

```suggestion
                    match issue_fs_freeze(app_node_endpoint.clone(), volume_uuid.to_string()).await
                    {
                        Err(error) if error.code() == Code::NotFound => Err(Status::not_found(format!(
                                "Failed to freeze volume {}, filesystem volume is not mounted",
                                volume_uuid
                            ))),
                        _else => _else
                    }?;
```
Also is the volume not mounted, or not staged/attached?

**Review by tiagolobocastro — APPROVED** [review]
This is a bit confusing:
`feat(fs const snap): return meaningful error if volume is not attached by fsfreeze flow is triggered`
It's also a fix, not a feature. Example:
`fix(fsfreeze): improved error message when volume is not staged`

File: control-plane/csi-driver/src/bin/controller/controller.rs (hunk 1 above)
Comment (Abhinandan-Purkait) [reply control-plane/csi-driver/src/bin/controller/controller.rs]: I wanted to use attached, was wondering if users would be more used to the term mounted than attached?

Comment (tiagolobocastro) [reply control-plane/csi-driver/src/bin/controller/controller.rs]: hmm not sure. Volume attachments is using the term attached, but they would have to know about it.

Comment (Abhinandan-Purkait) [reply control-plane/csi-driver/src/bin/controller/controller.rs]: Alright. I will use attached here.

Comment (Abhinandan-Purkait) [reply control-plane/csi-driver/src/bin/controller/controller.rs]: Done.

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**